students/tri_nguyen/lesson06/test_mailroom.py

```diff
@@ -0,0 +1,101 @@
+#!/usr/bin/env python3
+
+# ------------------------------------------- #
+# Student:   Tri Nguyen
+# Instructor: Natasha Aleksandrova
+# Date:  16-Mar-2018
+# ------------------------------------------- #
+
+import unittest
+import mailroom
+
+class TestMailroom(unittest.TestCase):
+
+    def test_display_menu(self):
+        ''' test display_menu function of mailroom.py '''
+
+        known_menu = ''' Select one of the actions below:
+
+    1. Send a Thank You
+    2. Create a Report
+    3. Send letters to everyone
+    4. Quit
+
+    '''
+
+        expected_result = mailroom.display_menu()
+        self.assertEqual(expected_result, known_menu)
+        # are assertIsNotNone and assertIsInstance necessary??
+        self.assertIsNotNone(expected_result)
+        self.assertIsInstance(expected_result, str)
+
+
+    def test_convert_to_float(self):
+        ''' test convert_to_float function of mailroom.py '''
+
+        self.assertEqual(mailroom.convert_to_float('100.00'), 100.0)
+        self.assertEqual(mailroom.convert_to_float('12.234'), 12.234)
+        self.assertEqual(mailroom.convert_to_float('145.12'), 145.12)
+        self.assertEqual(mailroom.convert_to_float('14.123456'), 14.123456)
+
+
+    def test_generate_email_template(self):
+        ''' test generate_email_template of mailroom.py '''
+
+        known_temp = '''
+                Dear {0},
+
+                Thank you for giving us ${1:,.2f}. Your money will be put to good use.
+                Please come back and donate more.
+
+                Best Regards,
+
+                John Doe
+            '''
+
+        expected_result = mailroom.generate_email_template()
+        self.assertEqual(expected_result, known_temp)
+        # are assertIsNotNone and assertIsInstance necessary??
+        self.assertIsNotNone(expected_result)
```

Comment:
Same as above, this is just testing the string assignment which is bordering on testing Python's built-in functionality. But you could make your `generate_email_template` function more testable by including `full_name and `donation_amount` parameters and then testing the formatted string output with different input values.